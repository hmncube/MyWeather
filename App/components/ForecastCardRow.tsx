import React from 'react';
import {View, StyleSheet, Text, Image} from 'react-native';
import darkTheme from '../constants/darkTheme';
const ForecastCardRow = ({image, title, value}) => {
  return (
    <View style={styles.container}>
      <View style={styles.imagesContainer}>
        <Image style={styles.smallImage} source={image} />
      </View>
      <View style={styles.textContainer}>
        <Text style={[styles.text, styles.textName]}>{title}</Text>
        <Text style={[styles.text, styles.texValue]}>{value}</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  imagesContainer: {
    flex: 1,
    margin: 4,
  },
  textContainer: {
    flex: 10,
    margin: 4,
    flexDirection: 'row',
  },
  container: {
    flexDirection: 'row',
  },
  text: {
    fontSize: 16,
    lineHeight: 21,
    fontWeight: 'bold',
    letterSpacing: 0.25,
    color: darkTheme.onSurface,
  },
  textName: {
    textAlign: 'left',
    flex: 1,
  },
  texValue: {
    marginEnd: 16,
  },
  smallImage: {
    height: 20,
    width: 20,
    textAlign: 'left',
  },
});

export default ForecastCardRow;
