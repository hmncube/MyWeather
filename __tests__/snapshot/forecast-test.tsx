import React from 'react';
import renderer from 'react-test-renderer';
import ForecastCardRow from '../../App/components/ForecastCardRow';

it('ForecastCardRow renders correctly', () => {
  const tree = renderer
    .create(
      <ForecastCardRow
        image={require('../../App/assets/images/hot.png')}
        title={'Test'}
        value={'Value'}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
