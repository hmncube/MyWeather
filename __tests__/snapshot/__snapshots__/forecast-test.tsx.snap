// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastCardRow renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "margin": 4,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../App/assets/images/temp.png",
        }
      }
      style={
        Object {
          "height": 20,
          "textAlign": "left",
          "width": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 10,
        "flexDirection": "row",
        "margin": 4,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          },
          Object {
            "flex": 1,
            "textAlign": "left",
          },
        ]
      }
    >
      Test
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": 0.25,
            "lineHeight": 21,
          },
          Object {
            "marginEnd": 16,
          },
        ]
      }
    >
      Value
    </Text>
  </View>
</View>
`;
